README.md

```diff
@@ -0,0 +1,403 @@
+```
+(base) ➜  bolt_terraform_demo bolt project init
+(base) ➜  bolt_terraform_demo ls
+bolt-project.yaml inventory.yaml
+```
+Bolt comes with a `terraform` module that will contain all the module code needed for this example. 
+```
+(base) ➜  bolt_terraform_demo bolt task show | grep terraform
+  terraform::apply                          Apply an HCL manifest
+  terraform::destroy                        Destroy resources managed with Terraform
+  terraform::initialize                     Initialize a Terraform project directory
+  terraform::output                         JSON representation of Terraform outputs
+```
+
+
+### Terraform HCL manifests
+
+I have created a very simple terraform manifest for this demo. The terraform code is stored in a dedicated `terraform` directory in my particular bolt project. This is where manifest code, variable refs and terraform state will be stored. 
+
+The main terraform manifest `main.tf` contains the declaration of terraform resources to provision some VMs in GCP. 
+```
+terraform {
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+      version = "4.40.0"
+    }
+  }
+}
+
+provider "google" {
+  project = var.project
+  region  = var.region
+  zone    = var.zone
+}
+
+data "google_compute_subnetwork" "subnet" {
+  name    = var.subnet
+  region  = var.region
+  project = var.subnet_project
+}
+
+locals {
+  network = data.google_compute_subnetwork.subnet.network
+  metadata = {
+    "ssh-keys" = "${var.user}:${file(var.ssh_key)}"
+  }
+}
+
+resource "google_compute_instance" "terraform_instance" {
+  name         = "terraform-instance-${count.index}"
+  count        = var.num_instances
+  machine_type = "e2-micro"
+
+  boot_disk {
+    initialize_params {
+      image = "debian-cloud/debian-11"
+    }
+  }
+
+  metadata = local.metadata
+
+  network_interface {
+    network            = local.network
+    subnetwork         = var.subnet
+    subnetwork_project = var.subnet_project
+  }
+}
+```
+
+The variables used to populate the manifest values are stored in `variables.tf`
+
+```
+variable "project" {
+  description = "Name of GCP project"
+  type        = string
+  default     = "team-skeletor-scratchpad"
+}
+
+variable "region" {
+  description = "GCP region that will be targeted for infrastructure deployment"
+  type        = string
+  default     = "us-west1"
+}
+
+variable "zone" {
+  description = "GCP zone that will be targeted for infrastructure deployment"
+  type        = string
+  default     = "us-west1-b"
+}
+
+variable "user" {
+  description = "User name associated with GCP"
+  type        = string
+  default     = "cas.donoghue"
+}
+
+variable "ssh_key" {
+  description = "Public ssh key to access GCP VMs"
+  type        = string
+  default     = "/Users/cas.donoghue/.ssh/id_rsa-gcloud.pub"
+}
+
+variable "subnet" {
+  description = "The subnet your project is on"
+  type        = string
+  default     = "team-skeletor-scratchpad"
```

Comment:
Is this stuff (subnet, subnet_project etc) private at all? (I realize its on a public git repo now, but should i take it out for the blog post).